tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+	'Fortitude', 
+	'Reflex',
+	'Will'
+);
+
+CREATE TYPE IF NOT EXISTS CharacterStat AS ENUM (
+	'Name',
+	'Race',
+	'Size',
+	'Height',
+	'Weight',
+	'Age',
+	'Gender',
+	'Alignment',
+	'Deity',
+	'Languages',
+	'Appearance'
+);
+
+CREATE TYPE IF NOT EXISTS CombatStat AS ENUM (
+	'Melee Attack Bonus',
+	'Ranged Attack Bonus',
+	'CMB',
+	'CMD',
+	'Armor Class',
+	'Touch AC',
+	'Flat-Footed AC',
+	'Initiative Bonus',
+	'Damage Reduction',
+	'Spell Resistance',
+	'Speed',
+	'Fortitude Save',
+	'Reflex Save',
+	'Will Save'
+);
+
+CREATE TYPE IF NOT EXISTS RaceType AS ENUM (
+	'Abberation', 
+	'Animal', 
+	'Construct', 
+	'Dragon', 
+	'Fey', 
+	'Humanoid', 
+	'Magical Beast', 
+	'Monstrous Humanoid', 
+	'Ooze', 
+	'Outsider', 
+	'Plant', 
+	'Undead', 
+	'Vermin'
+);
+
+CREATE TYPE IF NOT EXISTS RaceSubtyep AS ENUM (
+
+);
+
+CREATE TYPE IF NOT EXISTS CasterType AS ENUM (
+	'Spontaneous', 
+	'Prepared'
+);
+
+CREATE TYPE IF NOT EXISTS ComponentType AS ENUM (
+	'Somatic',
+	'Material',
+	'Verbal'
+);
+
+CREATE TYPE IF NOT EXISTS MagicSchool AS ENUM (
+	'Abjuration', 
+	'Conjuration', 
+	'Divination', 
+	'Enchantment', 
+	'Evocation', 
+	'Illusion', 
+	'Necromancy', 
+	'Transmutation'
+);
+
+CREATE TYPE IF NOT EXISTS SpellRange AS ENUM (
+	'Personal',
+	'Touch',
+	'Close',
+	'Medium',
+	'Long',
+	'Unlimited'
+);
+
+CREATE TYPE IF NOT EXISTS WeaponClass AS ENUM (
+	'Axes', 
+	'Heavy Blades', 
+	'Light Blades', 
+	'Bows', 
+	'Close', 
+	'Crossbows', 
+	'Double', 
+	'Firearms', 
+	'Flails', 
+	'Hammers', 
+	'Monk', 
+	'Natural', 
+	'Polearms', 
+	'Siege Engines', 
+	'Spears', 
+	'Thrown', 
+	'Tribal'
+);
+
+CREATE TYPE IF NOT EXISTS ArmorClass AS ENUM (
+	'Light', 
+	'Medium', 
+	'Heavy'
+);
+
+CREATE TYPE IF NOT EXISTS EquipmentSlot AS ENUM (
+	'None',
+	'Armor', 
+	'Belts', 
+	'Body', 
+	'Chest', 
+	'Eyes', 
+	'Feet', 
+	'Hands', 
+	'Head', 
+	'Headband', 
+	'Neck', 
+	'Ring (left)', 
+	'Ring (right)', 
+	'Shield', 
+	'Shoulders', 
+	'Wrist'
+);
+
+CREATE TYPE IF NOT EXISTS DamageType AS ENUM (
+	'Bludgeoning', 
+	'Slashing', 
+	'Piercing', 
+	'Energy', 
+	'Acid', 
+	'Fire', 
+	'Electricity', 
+	'Cold', 
+	'Sonic', 
+	'Positive', 
+	'Negative', 
+	'Nonlethal'
+);
+
+--table declarations
+--user creds table - also used for hashing info
+CREATE TABLE IF NOT EXISTS Users (
+	id     			UUID PRIMARY KEY,
+
+	email			TEXT 
+		NOT NULL,
+        username       		TEXT 
+		NOT NULL,
+	nickname		TEXT 
+		NOT NULL,
+        is_admin    		BOOL 
+		NOT NULL,
+
+        pass_hash   		BYTEA 
+		NOT NULL,
+        salt        		BYTEA 
+		NOT NULL,
+
+        time_cost   		INT 
+		NOT NULL,
+        memory      		INT 
+		NOT NULL,
+        threads     		INT 
+		NOT NULL
+);
+
+--Character overview table
+CREATE TABLE IF NOT EXISTS Characters (
+        char_id         	UUID PRIMARY KEY,
+
+	id			UUID REFERENCES Users 
+		NOT NULL,
+        race_id         	UUID REFERENCES Races 
+		NOT NULL,
+        deity_id        	UUID REFERENCES Deities 
+		NOT NULL,
```

Comment:
I believe you need to specify which column each foreign key references, not just the table.

Reply:
This is actually something I found out - if you only specify the table, not the foreign key, then Postgres immediately assumes it to be the primary key of the referenced table. Easy shorthand that sped up the writing process.